### PR TITLE
Class implementation

### PR DESCRIPTION
**Pull request by GordonOtieno**
**Refactored Awesome Books**
- added table with basic look and feel.
- implemented dynamic re-rendering of UI using classes.
-  Implemented delete and add methods in class.

### CODE REVIEW

File: awesome.js [hunk 1]
```diff
@@ -1,50 +1,63 @@
...
+
+  removeBook(index) {
+    if (index >= 0) {
+      const decodedbooks = JSON.parse(localStorage.getItem('books'));
+      decodedbooks.splice(index, 1);
+      localStorage.setItem('books', JSON.stringify(decodedbooks));
+      const update = JSON.parse(localStorage.getItem('books'));
+      this.displayrecord(update);
+    }
+  }
```
Comment (DammyShittu) [inline awesome.js]: - [ ] When there is only one book on the page and I click to remove it, it is removed in the local storage but still stays on the page. But that is not the case when there are multiple books on the page. As you can see the local storage is empty but there is a book on the page. Kindly fix that.

  next-book

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi team,

## Highlights
✔️ Descriptive PR
✔️ Passing linters

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by leonardodiasb — CHANGES_REQUESTED** [review]
# Required Changes ♻️ ⚠️ 🚦 

Hi @GordonOtieno, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

mario_ladders

## Good points 👍 

You used proper GitHub Flow ✔️
Your linter checks are all passing ✔️
Your README and PR message are descriptive ✔️

## Aspects to improve ♻️

- [ ] While you did great effort trying to apply the changes required by the last reviewer, something in your code broke. I cannot add any book to your list, it seems that the "Add" button does not have an event. Kindly fix it.
Untitledgif


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by GordonOtieno** [comment]
I cant figure out what you mean, I can add and delete successfully. please
elaborate

**Review by topeogunleye — APPROVED** [review]
### Hi @GordonOtieno, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Followed Github flow :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and description :heavy_check_mark:
 
## [Optional] suggestions:
* Nothing to mention.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**